src/NHibernate/Linq/Visitors/ParameterTypeLocator.cs

```diff
@@ -219,14 +222,35 @@ protected override Expression VisitConstant(ConstantExpression node)
 				return node;
 			}
 
-			public override Expression Visit(Expression node)
+			protected override Expression VisitSubQuery(SubQueryExpression node)
 			{
-				if (node is SubQueryExpression subQueryExpression)
+				// ReLinq wraps all ResultOperatorExpressionNodeBase into a SubQueryExpression. In case of
+				// ContainsResultOperator where the constant expression is dislocated from the related expression,
+				// we have to manually link the related expressions.
+				var containsOperator = node.QueryModel.ResultOperators.OfType<ContainsResultOperator>().FirstOrDefault();
```

Comment:
Can it be optimized to something like the following?
```C#
node.QueryModel.ResultOperators.Count == 1 && node.QueryModel.ResultOperators[0] is ContainsResultOperator containsOperator
```

Reply:
Done.